core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java

```diff
@@ -8280,6 +8283,15 @@ private void checkLiteral2(String expression, String expected) {
         .dialect(ClickHouseSqlDialect.DEFAULT).ok(clickhouseExpected2);
     sql(queryFalse1)
         .dialect(ClickHouseSqlDialect.DEFAULT).ok(clickhouseExpected2);
+
+    sql(queryTrue2).dialect(ClickHouseSqlDialect.DEFAULT)
+        .ok("SELECT (1 = 1)\nFROM `foodmart`.`employee`");
+    sql(queryTrue2).dialect(MssqlSqlDialect.DEFAULT)
+        .ok("SELECT (1 = 1)\nFROM [foodmart].[employee]");
+    sql(queryFalse2).dialect(ClickHouseSqlDialect.DEFAULT)
+        .ok("SELECT (1 = 0)\nFROM `foodmart`.`employee`");
+    sql(queryFalse2).dialect(MssqlSqlDialect.DEFAULT)
+        .ok("SELECT (1 = 0)\nFROM [foodmart].[employee]");
```

Comment:
As Jira https://issues.apache.org/jira/browse/CALCITE-6829, It is not a valid SQL in MSSQL.

Reply:
OK, I remove mssql sql of this case. @NobiGo